src/rational/poly_over_q/arithmetic/add.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright © 2023 Phil Milewski
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! Implementation of the [`Add`] trait for [`PolyOverQ`] values.
+
+use super::super::PolyOverQ;
+use crate::macros::arithmetics::{
+    arithmetic_trait_borrowed_to_owned, arithmetic_trait_mixed_borrowed_owned,
+};
+use flint_sys::fmpq_poly::fmpq_poly_add;
+use std::ops::Add;
+
+impl Add for &PolyOverQ {
+    type Output = PolyOverQ;
+    /// Implements the [`Add`] trait for two [`PolyOverQ`] values.
+    /// [`Add`] is implemented for any combination of [`PolyOverQ`] and borrowed [`PolyOverQ`].
+    ///
+    /// Parameters:
+    /// - `other`: specifies the value to add to `self`
+    ///
+    /// Returns the sum of both polynomials as a [`PolyOverQ`].
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::rational::PolyOverQ;
+    /// use std::str::FromStr;
+    ///
+    /// let a: PolyOverQ = PolyOverQ::from_str("3  1/8 2/7 -7").unwrap();
+    /// let b: PolyOverQ = PolyOverQ::from_str("5  1/9 2/9 -3/8 0 8/9").unwrap();
+    ///
+    /// let c: PolyOverQ = &a + &b;
+    /// let d: PolyOverQ = a + b;
+    /// let e: PolyOverQ = &c + d;
+    /// let f: PolyOverQ = c + &e;
+    /// ```
+    fn add(self, other: Self) -> Self::Output {
+        let mut out = PolyOverQ::default();
+        unsafe {
+            fmpq_poly_add(&mut out.poly, &self.poly, &other.poly);
+        }
+        out
+    }
+}
+
+arithmetic_trait_borrowed_to_owned!(Add, add, PolyOverQ, PolyOverQ, PolyOverQ);
+arithmetic_trait_mixed_borrowed_owned!(Add, add, PolyOverQ, PolyOverQ, PolyOverQ);
+
+#[cfg(test)]
+mod test_add {
+
+    use super::PolyOverQ;
+    use crate::rational::Q;
+    use std::str::FromStr;
+
+    /// testing addition for two [`PolyOverQ`]
+    #[test]
+    fn add() {
+        let a: PolyOverQ = PolyOverQ::from_str("3  1/7 2 -3").unwrap();
```

Comment:
you don't have to explicitely mention the type. Do as you like, but it is not needed. This goes for many of your tests in other PRs too. I just wanted to mention it once